mizar/dp/mizar/workflows/builtins/networkpolicies/create.py

```diff
@@ -103,3 +113,51 @@ def init_data_for_networkpolicy(self):
             "port_table": [],
         }
         return data
+
+    def fill_data_from_onegress(self, data, gress_type, networkpolicy):
+        network_policy_name = networkpolicy["metadata"]["name"]
+        for index, onegress in enumerate(networkpolicy["spec"][gress_type]):
+            indexed_policy_name = "{}_{}_{}".format(network_policy_name, gress_type, index)
+            if network_policy_name not in data["networkpolicy_map"]:
+                data["networkpolicy_map"][network_policy_name] = set()
+            if indexed_policy_name not in data["networkpolicy_map"][network_policy_name]:
+                data["networkpolicy_map"][network_policy_name].add(indexed_policy_name)
+                data["indexed_policy_count"] += 1
+
+            self.fill_cidrs_from_onegress(data, indexed_policy_name, gress_type, onegress)
+
+    def fill_cidrs_from_onegress(self, data, indexed_policy_name, gress_type, onegress):
+        if indexed_policy_name not in data["ports_map"]:
+            data["ports_map"][indexed_policy_name] = []
+        for port in onegress["ports"]:
+            data["ports_map"][indexed_policy_name].append("{}:{}".format(port["protocol"], port["port"]))
+
+        if indexed_policy_name not in data["cidrs_map_no_except"]:
+            data["cidrs_map_no_except"][indexed_policy_name] = []
+        if indexed_policy_name not in data["cidrs_map_with_except"]:
+            data["cidrs_map_with_except"][indexed_policy_name] = []
+        if indexed_policy_name not in data["cidrs_map_except"]:
+            data["cidrs_map_except"][indexed_policy_name] = []
+        for gress_item in onegress["from" if gress_type == "ingress" else "to"]:
+            if "ipBlock" in gress_item:
+                if "except" in gress_item["ipBlock"]:
+                    data["cidrs_map_with_except"][indexed_policy_name].append(gress_item["ipBlock"]["cidr"])
+                    for except_cidr in gress_item["ipBlock"]["except"]:
+                        data["cidrs_map_except"][indexed_policy_name].append(except_cidr)
+                else:
+                    data["cidrs_map_no_except"][
+                        indexed_policy_name].append(gress_item["ipBlock"]["cidr"])
+            elif "namespaceSelector" in gress_item and "podSelector" in gress_item:
+                raise NotImplementedError("Not implemented")
+            elif "namespaceSelector" in gress_item:
+                raise NotImplementedError("Not implemented")
+            elif "podSelector" in gress_item:
+                pods = list_pods_by_labels(gress_item["podSelector"]["matchLabels"])
+                for pod in pods.items:
+                    data["cidrs_map_no_except"][indexed_policy_name].append("{}/32".format(pod.status.pod_ip))
+            else:
+                raise NotImplementedError("Not implemented for {}".format(gress_item))
+
+    def build_gressdata(self, data, ep):
+        #TODO Build data that fits for daemon data format
```

Comment:
Same with the naming. please rename "gressdata"

Reply:
Fixed

Reply:
build_directional_data => build_access_rules

Reply:
Fixed